### DISCUSSION TITLE
how to make mobilenet_224 model work with 112x112 sized images?

### DISCUSSION THREAD

**Discussion by 112x112** [sonos/tract#1486] [Q&A] [answered]
hi

thanks for this repository

```rust
pub fn foo(s: &str) -> Tensor {
    let i = image::open(s).unwrap().to_rgb8();
    let resized = image::imageops::resize(&i, 112, 112, ::image::imageops::FilterType::Triangle);
    tract_ndarray::Array4::from_shape_fn((1, 112, 112, 3), |(_, y, x, c)| {
        f32::from(resized[(u32::try_from(x).unwrap(), u32::try_from(y).unwrap())][c]) / 255.0
    })
    .into()
}
```

im trying to use 112x112 images with

```rust
    tract_tflite::tflite()
        .model_for_path("mobilenet_v3_small_100_224.tflite")?
        .into_optimized()?
        .into_runnable()
```

but i get the error

```
called `Result::unwrap()` on an `Err` value: Clashing resolution for expression. 224=224 != 112.
```

i think the problem is that I'm trying to use 112x112 images with a 224 mobilenet model.

do you by any chance know how to make Mobilenet support 112x112 sized images?

**Comment by 112x112** [comment]
https://github.com/sonos/tract/discussions/1464 might be related

**Reply by kali** [reply]
Not really. IIRC, this was an actual tract bug, what you experiencing is a model / input mismatch.

**Comment by kali** [comment, marked as answer]
The input for this specific instance of mobilenet MUST be 224 squared. Period.

You can fish around for instance of mobilenet for different input sizes. If you can't find a 112x112 you may need to train one yourself.

Alternatively you may upscale your input images to 224x224, (docs.rs/image may help) but I have no idea how it will impact the model performance.

**Reply by 112x112** [reply]
ty!

> If you can't find a 112x112 you may need to train one yourself.

do you have any links to resources on how to do this?

**Reply by kali** [reply]
I don't, but mobilenet is not a recent model, there must be plenty of resources around.
